casper/src/test/scala/coop/rchain/casper/genesis/DeployPaymentCostTest.scala

```diff
@@ -0,0 +1,196 @@
+package coop.rchain.casper.genesis
+
+import cats.{Apply, Id}
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.casper.genesis.DeployPaymentCostTest._
+import coop.rchain.casper.genesis.contracts.{Faucet, PreWallet}
+import coop.rchain.casper.helper.HashSetCasperTestNode
+import coop.rchain.casper.protocol.{BlockMessage, DeployData}
+import coop.rchain.casper.util.ProtoUtil
+import coop.rchain.casper.util.rholang.InterpreterUtil
+import coop.rchain.casper.{Created, HashSetCasperTest, MultiParentCasperImpl, Valid}
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.{Blake2b256, Keccak256}
+import coop.rchain.crypto.signatures.{Ed25519, Secp256k1}
+import coop.rchain.models.Expr.ExprInstance.{GByteArray, GInt}
+import coop.rchain.models.Var.VarInstance.{BoundVar, FreeVar}
+import coop.rchain.models._
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.rholang.interpreter.{accounting, Interpreter}
+import coop.rchain.rspace.Serialize
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.FlatSpec
+
+import scala.collection.immutable.BitSet
+
+class DeployPaymentCostTest extends FlatSpec {
+
+  "DeployPaymentCost" should "estimate cost of deploying payment code" in {
```

Comment:
This test is difficult to follow. Can you refactor it in order to improve its readability?

Reply:
It will be difficult to follow because it mixes various approached - handcrafted rholang, parsed rholang, deploying blocks (using Casper's API) and Registry. I will try to improve it but don't expect too much.